src/interfaces/IAppModule.ts

```diff
@@ -0,0 +1,42 @@
+import { CollectionModule } from "../modules/collection";
+import { DatastoreModule } from "../modules/datastore";
+import { DropModule } from "../modules/drop";
+import { MarketModule } from "../modules/market";
+import { NFTModule } from "../modules/nft";
+import { PackModule } from "../modules/pack";
+import { SplitsModule } from "../modules/royalty";
+import { CurrencyModule } from "../modules/token";
+import CollectionModuleMetadata from "../types/module-deployments/CollectionModuleMetadata";
+import CurrencyModuleMetadata from "../types/module-deployments/CurrencyModuleMetadata";
+import DatastoreModuleMetadata from "../types/module-deployments/DatastoreModuleMetadata";
+import DropModuleMetadata from "../types/module-deployments/DropModuleMetadata";
+import MarketModuleMetadata from "../types/module-deployments/MarketModuleMetadata";
+import NftModuleMetadata from "../types/module-deployments/NftModuleMetadata";
+import PackModuleMetadata from "../types/module-deployments/PackModuleMetadata";
+import SplitsModuleMetadata from "../types/module-deployments/SplitsModuleMetadata";
+
+export default interface IAppModule {
+  deployCollectionModule(
+    metadata: CollectionModuleMetadata,
+  ): Promise<CollectionModule>;
+
+  deploySplitsModule(metadata: SplitsModuleMetadata): Promise<SplitsModule>;
+
+  deployNftModule(metadata: NftModuleMetadata): Promise<NFTModule>;
+
+  deployCurrencyModule(
+    metadata: CurrencyModuleMetadata,
+  ): Promise<CurrencyModule>;
+
+  deployMarketplaceModule(
+    metadata: MarketModuleMetadata,
+  ): Promise<MarketModule>;
+
+  deployPackModule(metadata: PackModuleMetadata): Promise<PackModule>;
+
+  deployDropModule(metadata: DropModuleMetadata): Promise<DropModule>;
```

Comment:
is it Drops or Drop?

I think in general we have a little bit of inconsistent between drop vs drops, split vs splits. We should settle on one

Reply:
`Drop` makes sense, `Splits` makes sense, `Drops` doesn't make sense to me, its an *NFT Drop*, not an *NFT Drops*

Reply:
i think there's a question on the product level. our sdk should match the branding of what we sell to our customers.
thirdweb.com and thirdweb.com/dashboard, we've branded it as `Drops` rather than `Drop`.

Reply:
Actually I think we're using DropModule in somewhere else, i think we should be fine. as long as we keep it consistent lol

Reply:
make it all singular.

Reply:
everything is singular, no more plurals unless it's explicitly an array then *maybe* a plural